allennlp/commands/fine_tune.py

```diff
@@ -0,0 +1,202 @@
+                               type=str,
+                               help='path to validation data to use for continuing training')
+
+        subparser.add_argument('-s', '--serialization-dir',
+                               required=True,
+                               type=str,
+                               help='directory in which to save the fine-tuned model and its logs')
+
+        subparser.add_argument('-o', '--overrides',
+                               type=str,
+                               default="",
+                               help='a HOCON structure used to override the experiment configuration')
+
+        subparser.add_argument('--include-package',
+                               type=str,
+                               action='append',
+                               default=[],
+                               help='additional packages to include')
+
+        subparser.add_argument('--file-friendly-logging',
+                               action='store_true',
+                               default=False,
+                               help='outputs tqdm status on separate lines and slows tqdm refresh rate')
+
+        subparser.set_defaults(func=fine_tune_model_from_args)
+
+        return subparser
+
+
+def fine_tune_model_from_args(args: argparse.Namespace):
+    """
+    Just converts from an ``argparse.Namespace`` object to string paths.
+    """
+    # Import any additional modules needed (to register custom classes)
+    for package_name in args.include_package:
+        import_submodules(package_name)
+    fine_tune_model_from_file_paths(model_archive_path=args.model_archive,
+                                    train_data_path=args.data_path,
+                                    serialization_dir=args.serialization_dir,
+                                    overrides=args.overrides,
+                                    file_friendly_logging=args.file_friendly_logging)
+
+
+def fine_tune_model_from_file_paths(model_archive_path: str,
+                                    train_data_path: str,
+                                    serialization_dir: str,
+                                    validation_data_path: str = None,
+                                    overrides: str = "",
+                                    file_friendly_logging: bool = False) -> Model:
+    """
+    A wrapper around :func:`fine_tune_model` which loads the model archive from a file.
+
+    Parameters
+    ----------
+    model_archive_path : ``str``
+        Path to a saved model archive that is the result of running the ``train`` command.
+    data_path : ``str``
+        The training data to use for continuing training the saved model.  We just pass this along
+        to :func:`fine_tune_model`.
+    serialization_dir : ``str``
+        The directory in which to save results and logs. We just pass this along to
+        :func:`fine_tune_model`.
+    validation_data_path : ``str``, optional
```

Comment:
While validation is optional, without it, I get this error: 

```
Traceback (most recent call last):
  File "/usr/local/Cellar/python3/3.6.4_2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/local/Cellar/python3/3.6.4_2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/daniel/ideaProjects/allennlp/allennlp/run.py", line 18, in <module>
    main(prog="python -m allennlp.run")
  File "/Users/daniel/ideaProjects/allennlp/allennlp/commands/__init__.py", line 65, in main
    args.func(args)
  File "/Users/daniel/ideaProjects/allennlp/allennlp/commands/fine_tune.py", line 87, in fine_tune_model_from_args
    file_friendly_logging=args.file_friendly_logging)
  File "/Users/daniel/ideaProjects/allennlp/allennlp/commands/fine_tune.py", line 125, in fine_tune_model_from_file_paths
    file_friendly_logging=file_friendly_logging)
  File "/Users/daniel/ideaProjects/allennlp/allennlp/commands/fine_tune.py", line 192, in fine_tune_model
    metrics = trainer.train()
  File "/Users/daniel/ideaProjects/allennlp/allennlp/training/trainer.py", line 673, in train
    self._update_learning_rate(epoch, val_metric=this_epoch_val_metric)
  File "/Users/daniel/ideaProjects/allennlp/allennlp/training/trainer.py", line 591, in _update_learning_rate
    raise ConfigurationError("The reduce_on_plateau learning rate scheduler requires "
allennlp.common.checks.ConfigurationError: 'The reduce_on_plateau learning rate scheduler requires a validation metric to compute the schedule and therefore must be used with a validation dataset.'
```

Reply:
Related to this: I am having some issues setting the validation set from commandline. Gave up trying and hard-coded it in the code.

Reply:
FYI: validation is optional in general, but your experiment seems to be using the `reduce_on_plateau` learning rate scheduler, which requires a validation set to work

Reply:
Ya that's what I figured after waiting ~20mins for to see this error. Ideally, the system should be smart to get this (say, with a clear error msg, right at the beginning).

Reply:
Unfortunately, that's a harder one for us to catch - we don't know what learning rate scheduler you're using, or whether it depends on the validation data.

For the problem setting the validation data from the command line, do you have a specific error?

Reply:
> Unfortunately, that's a harder one for us to catch - we don't know what learning rate scheduler you're using, or whether it depends on the validation data. 

Hmm ... I was assuming that this should be easy to do. If not, nvm.